jib-core/src/main/java/com/google/cloud/tools/jib/http/FailoverHttpClient.java

```diff
@@ -209,10 +221,18 @@ public Response put(URL url, Request request) throws IOException {
    * @throws IOException if building the HTTP request fails.
    */
   public Response call(String httpMethod, URL url, Request request) throws IOException {
-    if (!isHttpsProtocol(url) && !enableHttpAndInsecureFailover) {
+    if (!isHttpsProtocol(url)) {
+      if (enableHttpAndInsecureFailover) {
+        return call(httpMethod, url, request, getHttpTransport(true));
+      }
       throw new SSLException("insecure HTTP connection not allowed: " + url);
     }
 
+    Optional<Response> optionalResponse = followFailoverHistory(httpMethod, url, request);
```

Comment:
Is there an issue here that we are following failover history even if enableHttpAndInsecureFailover is false?

Reply:
`enableHttpAndInsecureFailover` false, then no failover happens and no history will ever be recorded, so `followFailoverHistory` will always return `Optional.empty()` immediately. I think it has no overhead, and this keeps the code simple. But one could surely argue if we add a branch for the no-failover-ever setup, it might increase readability.

Reply:
oh I see, no this is fine. I just needed to look at it again.